src/Patterns/ProductUpdater.php

```diff
@@ -249,9 +249,18 @@ public function update_product_content( $ai_generated_product_content ) {
 			return;
 		}
 
-		$product->set_name( $ai_generated_product_content['title'] );
-		$product->set_description( $ai_generated_product_content['description'] );
-		$product->set_regular_price( $ai_generated_product_content['price'] );
+		wp_update_post(
+			array(
+				'ID'           => $product->get_id(),
+				'post_title'   => $ai_generated_product_content['title'],
+				'post_content' => $ai_generated_product_content['description'],
+				'post_name'    => sanitize_title( $ai_generated_product_content['title'] ),
+				'meta_input'   => array(
+					'_regular_price' => $ai_generated_product_content['price'],
+				),
+			)
+		);
+		flush_rewrite_rules();
```

Comment:
@albarin, my apologies, as this just occurred to me after the PR was merged: could we change here to ensure the rewrite rules are flushed only after all dummy products were created? Flushing the rewrite rules every time a new product is created is very expensive.